resources/service.rb

```diff
@@ -22,22 +22,32 @@
           default: lazy { default_syslog_ng_service_name },
           description: 'The service name to perform actions upon'
 
+property :config_file, String,
+          default: lazy { "#{syslog_ng_config_dir}/syslog-ng.conf" },
+          description: 'The path to the Syslog-NG server configuration on disk'
+
 property :config_test, [true, false],
           default: true,
           description: 'Perform a configuration file test before performing service action'
 
 action_class do
   def do_service_action(resource_action)
-    if new_resource.config_test && %w(start restart reload).include?(resource_action)
-      command_path = platform_family?('amazon') ? '/sbin/syslog-ng' : '/usr/sbin/syslog-ng'
-      execute 'syslog-ng-config-test' do
-        command "#{command_path} -s"
-        action :run
-      end
-    end
+    log new_resource.config_test.to_s
```

Comment:
We probably don’t want to leave the log resource in. Chef 15 users will have this trigger on every run